Implement Committees Page

harsh3dev opened the pull request:
<!-- Thanks for contributing to OWASP Nest!-->

<!-- Don't forget to link your PR to an existing issue if any.-->
Resolves #91 

<!-- Describe the big picture of your changes.-->
This pull request introduces a new `Committees` page to the application, along with several related updates. The changes include adding the `Committees` page to the routing, modifying components to support the new page, and making minor style adjustments.
<!-- Thanks again for your contribution!-->

arkid15r commented on frontend/src/pages/Committees.tsx:
```diff
@@ -0,0 +1,58 @@
+    const [committeeData, setCommitteeData] = useState<CommitteeDataType | null>(null)
+
+    useEffect(() => {
+      document.title = 'OWASP Projects'
```
```suggestion
      document.title = 'OWASP Committees'
```

arkid15r commented on frontend/src/pages/Committees.tsx:
```diff
@@ -0,0 +1,58 @@
+      <div className="flex min-h-screen w-full flex-col items-center justify-normal p-5 text-text md:p-20">
+        <div className="flex h-fit w-full flex-col items-center justify-normal gap-4">
+          {committeeData &&
+            committeeData.committees.map((project, index) => {
```
Here and below
```suggestion
            committeeData.committees.map((committee, index) => {
```

arkid15r commented on frontend/src/pages/Committees.tsx:
```diff
@@ -0,0 +1,58 @@
+
+              const SubmitButton = {
+                label: 'Learn More',
+                icon: <FontAwesomeIconWrapper icon="fa-solid fa-code" />,
```
```suggestion
                icon: <FontAwesomeIconWrapper icon="fa-solid fa-people-group" />,
```

arkid15r reviewed (commented):
One more page 👍 

Please look into the comments below:

arkid15r reviewed (commented):
I've just also added coverage thresholds for jest as we want to keep our test coverage on a decent level.

Could you add committees pages tests similar to projects page?

harsh3dev commented:
> I've just also added coverage thresholds for jest as we want to keep our test coverage on a decent level.
> 
> Could you add committees pages tests similar to projects page?

Sure I will do it and update the PR

harsh3dev commented:
I have added the tests and also made the suggested changes:
![image](https://github.com/user-attachments/assets/a0409c9d-86f1-47f2-b693-cc547e32d477)

kasya commented on frontend/src/components/Card.tsx:
Also, could you update this component on line [151](https://github.com/OWASP/Nest/pull/156/files#diff-5e1fda4ddb8adb886b9b3e6e9a7bc05938b3433de88a94e9e0d0a69bf3796bafL151) to be:
`{social && social.length > 0 && (`

Since social is a list we also need to make sure the length is > 0.

harsh3dev commented:
@kasya 
I have updated the social icons and added `social.length > 0` validation as well.
Thanks for reviewing my code.
![image](https://github.com/user-attachments/assets/74a015dc-7670-4298-b0e5-91d9f580b6f7)

arkid15r commented:
@harsh3dev could you look into the new pre-commit warnings we have?
Thank you!

harsh3dev commented:
@arkid15r I have handled the warnings.
@kasya Please have a look at the changes I made as you have suggested.
Thank You

arkid15r reviewed (commented):
LGTM 👍

kasya reviewed (approved):
Thanks for addressing my comments @harsh3dev 
I also pushed some changes regarding Card component style. 

LGTM! Great contribution! 👍🏼
